src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -638,6 +638,54 @@ export default class RouteModel extends BaseModel {
         return false;
     }
 
+    /**
+     * Returns whether the specified runway is valid for this route's SID leg (if it has one)
+     *
+     * If there is no SID, there is no issue with changing runways, so we would treat this as "valid"
+     *
+     * @for RouteModel
+     * @method isRunwayModelValidForSid
+     * @param runwayModel {RunwayModel}
+     * @return {boolean}
+     */
+    isRunwayModelValidForSid(runwayModel) {
+        const sidLegIndex = this._findSidLegIndex();
+        const sidLegModel = this._legCollection[sidLegIndex];
+
+        if (!sidLegModel) {
+            return true;
+        }
+
+        const departureAirportIcao = this.getDepartureRunwayAirportIcao().toUpperCase();
+        const entryName = `${departureAirportIcao}${runwayModel.name}`;
+
+        return sidLegModel.procedureHasEntry(entryName);
+    }
+
+    /**
+     * Returns whether the specified runway is valid for this route's STAR leg (if it has one)
+     *
+     * If there is no STAR, there is no issue with changing runways, so we would treat this as "valid"
+     *
+     * @for RouteModel
+     * @method isRunwayModelValidForStar
+     * @param runwayModel {RunwayModel}
+     * @return {boolean}
+     */
+    isRunwayModelValidForStar(runwayModel) {
```

Comment:
I'd encourage you to see where these two methods could share logic.  there is a fair amount of repeated code between these two methods.

Reply:
I guess they're kinda parallel, but I wouldn't say they're that similar... they're using different methods in like 3 different places. Plus, I can't even imagine a sensible name for a method I could abstract that out to, and my thought is that if it's too abstract to have a sensible name, it probably does more harm than good...